closed/src/java.base/share/classes/sun/security/provider/NativeDigest.java

```diff
@@ -55,10 +57,25 @@
     //  0: is already reset
     private long bytesProcessed;
 
-    private static NativeCrypto nativeCrypto;
+    final private static NativeCrypto nativeCrypto;
+    final private static Cleaner digestCleaner;
 
     static {
         nativeCrypto = NativeCrypto.getNativeCrypto();
+        digestCleaner = CleanerFactory.cleaner();
+    }
+
+    private static class DigestCleanerRunnable implements Runnable {
+        private long digestCtx;
```

Comment:
Please apply the same fixes as were made for jdk8. Here, make the class and field `final`.